src/test/java/com/shapesecurity/shift/es2017/codegen/CodeGenTest.java

```diff
@@ -587,6 +587,8 @@ public void testLiteralNumericExpression() throws JsError {
         test("1e308", "1e+308");
         test("1e308");
         test("1e-308");
+        test("9223372036854775807");
```

Comment:
```
$ node
> 9223372036854775807 < 9223372036854775808
false
```

You should choose another double that's actually smaller than `2 ** 63` to test this. I believe this would just work even without your fix. `9223372036854775295` seems to be the largest double that's smaller than `2 ** 63`.